src/server/controllers/s3.js

```diff
@@ -12,6 +13,14 @@ const defaultConfig = {
 module.exports = function s3 (config) {
   config = Object.assign({}, defaultConfig, config)
 
+  // Use credentials to allow assumed roles to pass STS sessions in.
+  // If the user doesn't specify key and secret, the default credentials (process-env)
+  // will be used by S3 in calls below.
+  let credentials
+  if ('key' in config && 'secret' in config) {
```

Comment:
might be better for us to do `if (config.key && config.secret) {` instead. Becase if we have:

```js
config = {key: null, secret: undefined}

'key' in config; // true
'secret' in config; // true
'undeclaredkey' in config; // false
```

Reply:
Agreed. WIll fix.